google/resource_cloudfunctions_function_test.go

```diff
@@ -0,0 +1,476 @@
+						funcResourceName, &function),
+					resource.TestCheckResourceAttr(funcResourceName,
+						"available_memory_mb", "128"),
+					testAccCloudFunctionsFunctionSource(fmt.Sprintf("gs://%s/index.zip", bucketName), &function),
+					testAccCloudFunctionsFunctionTrigger(FUNCTION_TRIGGER_TOPIC, &function),
+					resource.TestCheckResourceAttr(funcResourceName,
+						"timeout", "61"),
+					resource.TestCheckResourceAttr(funcResourceName,
+						"entry_point", "helloPubSub"),
+					resource.TestCheckResourceAttr(funcResourceName,
+						"trigger_topic", topicName),
+				),
+			},
+			{
+				ResourceName:      funcResourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+func TestAccCloudFunctionsFunction_bucket(t *testing.T) {
+	t.Parallel()
+
+	var function cloudfunctions.CloudFunction
+
+	funcResourceName := "google_cloudfunctions_function.function"
+	functionName := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+	bucketName := fmt.Sprintf("tf-test-bucket-%d", acctest.RandInt())
+	zipFilePath, err := createZIParchiveForIndexJs(testBucketTriggerPath)
+	if err != nil {
+		t.Errorf(err.Error())
+		t.FailNow()
+	}
+	defer os.Remove(zipFilePath) // clean up
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudFunctionsFunctionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudFunctionsFunction_bucket(functionName, bucketName, zipFilePath),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCloudFunctionsFunctionExists(
+						funcResourceName, &function),
+					resource.TestCheckResourceAttr(funcResourceName,
+						"available_memory_mb", "128"),
+					testAccCloudFunctionsFunctionSource(fmt.Sprintf("gs://%s/index.zip", bucketName), &function),
+					testAccCloudFunctionsFunctionTrigger(FUNCTION_TRIGGER_BUCKET, &function),
+					resource.TestCheckResourceAttr(funcResourceName,
+						"timeout", "61"),
+					resource.TestCheckResourceAttr(funcResourceName,
+						"entry_point", "helloGCS"),
+					resource.TestCheckResourceAttr(funcResourceName,
+						"trigger_bucket", bucketName),
+				),
+			},
+			{
+				ResourceName:      funcResourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckCloudFunctionsFunctionDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*Config)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_cloudfunctions_function" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		project := rs.Primary.Attributes["project"]
+		region := rs.Primary.Attributes["region"]
+		cloudFuncId := &cloudFunctionId{
+			Project: project,
+			Region:  region,
+			Name:    name,
+		}
+		_, err := config.clientCloudFunctions.Projects.Locations.Functions.Get(cloudFuncId.cloudFunctionId()).Do()
+		if err == nil {
+			return fmt.Errorf("CloudFunctions still exists")
```

Comment:
nit: CloudFunctions -> Function (since you're saying that a function is still there, rather than the product as a whole)

Reply:
Fixed